benchmarks/generate_charts.py

```
@@ -0,0 +1,839 @@
+    grp.add_argument("--input", metavar="JSON",
+                     help="Path to *_raw.json from a real run")
+    parser.add_argument("--output", metavar="DIR",
+                        default="benchmarks/results",
+                        help="Output directory (default: benchmarks/results)")
+    args = parser.parse_args()
+
+    if args.input:
+        bdata = load_from_json(args.input)
+        print(f"Loaded {len(bdata.tasks)} tasks from {args.input}")
+    else:
+        bdata = sample_data()
```

Ruff (0.15.7)

[error] Undefined name `sample_data`

(F821)